### PR TITLE
[move][vm] Remove resource specialization in VM values

### PR DESCRIPTION
**Pull request by tnowacki**
- Remove resource variants for structs and vectors in VM values to make way for ability/constraint system
- Cleaned up related code

## Motivation

- While having low cost, additional levels of safety for resources is nice, it is not really feasible in the new constraint+ability system
- This cleanup is to make that PR easier

## Test Plan

- cargo test

## Related PRs

#7376

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: language/move-vm/types/src/values/values_impl.rs [hunk 1]
```diff
@@ -2011,21 +1870,13 @@ impl Vector {
...
-                    ))))
-                } else {
-                    Value(ValueImpl::Container(Container::VecC(Rc::new(
-                        RefCell::new(vec![]),
-                    ))))
-                }
-            }
+            Type::Vector(_) | Type::Struct(_) | Type::StructInstantiation(_, _) => Value(
+                ValueImpl::Container(Container::Vec(Rc::new(RefCell::new(vec![])))),
+            ),
```
Comment (vgao1996) [inline language/move-vm/types/src/values/values_impl.rs:1879]: These two cases can be merged.

**Review by sblackshear — APPROVED** [review]
Love those red diffs!

**Review by vgao1996 — APPROVED** [review]
Also LGTM!

**Pull request closed without merging**